### PR TITLE
Add accout.signRaw function to sign a message without prefix

### LINKED ISSUE #6955: Add raw hash signing function (without the wrapping in Ethereum message: ...)

### PR DESCRIPTION
**Pull request by blackmoshui**
## Description

Fix #6955

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth-accounts/src/account.ts [hunk 1]
```diff
@@ -154,9 +155,13 @@ export const parseAndValidatePrivateKey = (data: Bytes, ignoreLength?: boolean):
...
  *
  * > "0x8144a6fa26be252b86456491fbcd43c1de7e022241845ffea1c3df066f7cfede"
+ *
+ * web3.eth.accounts.hashMessage("Hello world", true)
+ *
+ * > "0xed6c11b0b5b808960df26f5bfc471d04c1995b0ffd2055925ad1be28d6baadfd"
  * ```
  */
-export const hashMessage = (message: string): string => {
+export const hashMessage = (message: string, skipPrefix?: boolean): string => {
```
Comment (jdevcs) [inline packages/web3-eth-accounts/src/account.ts]: above will work but using a default parameter is generally preferable in this case
```suggestion
export const hashMessage = (message: string, skipPrefix: boolean = false): string => {
```

File: packages/web3-eth-accounts/src/account.ts [hunk 2]
```diff
@@ -154,9 +155,13 @@ export const parseAndValidatePrivateKey = (data: Bytes, ignoreLength?: boolean):
  * web3.eth.accounts.hashMessage(web3.utils.utf8ToHex("Hello world")) // Will be hex decoded in hashMessage
  *
  * > "0x8144a6fa26be252b86456491fbcd43c1de7e022241845ffea1c3df066f7cfede"
+ *
+ * web3.eth.accounts.hashMessage("Hello world", true)
```
Comment (jdevcs) [inline packages/web3-eth-accounts/src/account.ts:159]: plus also need to update [changelog](https://github.com/web3/web3.js/blob/4.x/packages/web3-eth-accounts/CHANGELOG.md) for this package

**Review by jdevcs — APPROVED** [review]

**Review by jdevcs — COMMENTED** [review]
also need update of https://github.com/blackmoshui/web3.js/blob/fix-6955-add-raw-sign-function/packages/web3-eth-accounts/CHANGELOG.md

**Comment by jdevcs** [comment]
@blackmoshui thanks

**Comment by blackmoshui** [comment]
> also need update of https://github.com/blackmoshui/web3.js/blob/fix-6955-add-raw-sign-function/packages/web3-eth-accounts/CHANGELOG.md

Added. 😄 Thanks for the reminder.

**Review by avkos — APPROVED** [review]

**Review by danforbes — APPROVED** [review]

**Comment by jdevcs** [comment]
@blackmoshui before merging this PR, there are merge conflicts needs to be fixed, thanks

**Comment by blackmoshui** [comment]
> @blackmoshui before merging this PR, there are merge conflicts needs to be fixed, thanks

Updated.

**Pull request merged**